src/main/java/de/tum/in/www1/artemis/repository/iris/IrisHestiaSessionRepository.java

```diff
@@ -0,0 +1,41 @@
+package de.tum.in.www1.artemis.repository.iris;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import de.tum.in.www1.artemis.domain.iris.session.IrisHestiaSession;
+
+/**
+ * Repository interface for managing {@link IrisHestiaSession} entities.
+ * Provides custom queries for finding hestia sessions based on different criteria.
+ */
+public interface IrisHestiaSessionRepository extends JpaRepository<IrisHestiaSession, Long> {
+
+    /**
+     * Finds a list of {@link IrisHestiaSession} based on the exercise and user IDs.
+     *
+     * @param codeHintId The ID of the code hint.
+     * @return A list of hestia sessions sorted by creation date in descending order.
+     */
+    @Query("""
+            SELECT s
+            FROM IrisHestiaSession s
+            WHERE s.codeHint.id = :codeHintId
+            ORDER BY s.creationDate DESC
+            """)
+    List<IrisHestiaSession> findByCodeHintId(Long codeHintId);
```

Comment:
```suggestion
    List<IrisHestiaSession> findByCodeHintIdOrderByCreationDateDesc(long codeHintId);
```
Spring should be able to derive this query.